src/profile-logic/profile-data.js

```diff
@@ -1027,6 +1027,53 @@ export function filterThreadToSearchString(
   );
 }
 
+export function filterThreadByTab(
+  thread: Thread,
+  relevantPages: Set<InnerWindowID>
+): Thread {
+  return timeCode('filterThreadByTab', () => {
+    if (relevantPages.size === 0) {
+      return thread;
+    }
+
+    const { frameTable, stackTable } = thread;
+
+    const frameMatchesFilterCache = new Map();
+    function frameMatchesFilter(frame) {
+      let result = frameMatchesFilterCache.get(frame);
```

Comment:
It's also probably a good idea to add an implicit type to a `let` variable. It takes a lot of reasoning to figure this one out here, especially since `Map` is also not typed.